widget/embedded/src/utils/quote.ts

```diff
@@ -326,3 +288,62 @@ export const getUniqueBlockchains = (steps: Step[]) => {
 
   return result;
 };
+
+export const sortQuotesBy = (
+  strategy: PreferenceType,
+  quotes: MultiRouteResponse['results']
+): MultiRouteResponse['results'] => {
+  return quotes.sort((quote1, quote2) => {
+    const route1Score =
+      quote1.scores?.find((score) => score.preferenceType === strategy)
+        ?.score ?? 0;
+    const route2Score =
+      quote2.scores?.find((score) => score.preferenceType === strategy)
+        ?.score ?? 0;
+
+    return route2Score - route1Score;
+  });
```

Comment:
If two quotes share an equal score, they are to be sorted according to their requestId :
```ts
 return quotes.sort((quote1, quote2) => {
    const getScore = (route: MultiRouteSimulationResult) =>
      route.scores?.find((score) => score.preferenceType === strategy)?.score ?? 0;

    const quote1Score = getScore(quote1);
    const quote2Score = getScore(quote2);

    if (quote1Score !== quote2Score) {
      return quote2Score - quote1Score;
    }

    const lowerQuoteId1 = quote1.requestId.toLowerCase();
    const lowerQuoteId2 = quote2.requestId.toLowerCase();

    return lowerQuoteId1.localeCompare(lowerQuoteId2);
  });
```

Reply:
fixed